### DISCUSSION TITLE
Power of 2 Int consts translates to Real type

### DISCUSSION THREAD

**Discussion by jklmnn** [Z3Prover/z3#5624] [Q&A]
I noticed that using the power of 2 Int type creates a Real type:
```Python
>>> A = Int('A')
>>> B = Int('B')
>>> A + (A ** B)
ToReal(A) + A**B
>>> A + ToInt(A ** B)
A + ToInt(A**B)
```
I can observe this behavior in 4.8.10 and 4.8.12 but not 4.8.9. It's both notable in the Python and the C API. My expectation would be that the result of this operation would be an Int type.
Is this intentional or is this a bug?

**Comment by NikolajBjorner** [comment]
It is intentional. For example -1 is an int and 2^-1 is 1/2. The number 1/2 is not an int.
